components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/request/PostAuthnMissingChallengeQuestionsHandler.java

```diff
@@ -92,18 +92,21 @@ public PostAuthnHandlerFlowStatus handle(HttpServletRequest httpServletRequest,
                                              AuthenticationContext authenticationContext)
             throws PostAuthenticationFailedException {
 
-        String forceChallengeQuestionConfig = "";
-
         if (log.isDebugEnabled()) {
             log.debug("Post authentication handling for missing security questions has started");
         }
-        try {
-            forceChallengeQuestionConfig = getResidentIdpProperty(getAuthenticatedUser(authenticationContext).
-                    getTenantDomain(), IdentityRecoveryConstants.ConnectorConfig.FORCE_ADD_PW_RECOVERY_QUESTION);
-        } catch (Exception e) {
-            log.error("Error occurred while retrieving resident IDP property for forced challenge questions", e);
+
+        if (authenticationContext == null || authenticationContext.getSequenceConfig() == null
+                || authenticationContext.getSequenceConfig().getAuthenticatedUser() == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Authentication context or sequence config or authenticated user is null.");
+            }
+            return PostAuthnHandlerFlowStatus.UNSUCCESS_COMPLETED;
         }
 
+        String forceChallengeQuestionConfig = getResidentIdpProperty(getAuthenticatedUser(authenticationContext)
```

Comment:
Should use SP tenant domain instead of user tenant domain?

Reply:
Yes. It should be SP tenant domain.